x-pack/plugins/graph/public/helpers/use_graph_loader.ts

```diff
@@ -52,60 +55,105 @@ export const useGraphLoader = ({ toastNotifications, coreStart }: UseGraphLoader
+            'xpack.graph.inspectAdapter.graphExploreRequest.description',
+            {
+              defaultMessage: 'This request queries Elasticsearch to fetch the data for the Graph.',
+            }
+          ),
+        }
+      );
+      inspectRequest.stats({
+        indexPattern: {
+          label: i18n.translate(
+            'xpack.graph.inspectAdapter.graphExploreRequest.dataView.description.label',
+            { defaultMessage: 'Data view' }
+          ),
+          value: indexName,
+          description: i18n.translate(
+            'xpack.graph.inspectAdapter.graphExploreRequest.dataView.description',
+            { defaultMessage: 'The data view that connected to the Elasticsearch indices.' }
+          ),
+        },
+      });
+      return inspectRequest;
+    },
+    [requestAdapter]
+  );
+
   // Replacement function for graphClientWorkspace's comms so
   // that it works with Kibana.
   const callNodeProxy = useCallback(
     (indexName: string, query: ExploreRequest, responseHandler: GraphExploreCallback) => {
-      const request = {
-        body: JSON.stringify({
-          index: indexName,
-          query,
-        }),
-      };
+      const dsl = { index: indexName, query };
+      const request = { body: JSON.stringify(dsl) };
       setLoading(true);
+
+      requestAdapter.reset();
+      const inspectRequest = getRequestInspector(indexName);
+      inspectRequest.json(dsl);
+
       return coreStart.http
-        .post<{ resp: { timed_out: unknown } }>('../api/graph/graphExplore', request)
+        .post<{ resp: estypes.GraphExploreResponse }>('../api/graph/graphExplore', request)
         .then(function (data) {
           const response = data.resp;
+
           if (response?.timed_out) {
             toastNotifications.addWarning(
               i18n.translate('xpack.graph.exploreGraph.timedOutWarningText', {
                 defaultMessage: 'Exploration timed out',
               })
             );
           }
+          inspectRequest.stats({}).ok({ json: response });
           responseHandler(response);
         })
-        .catch(handleHttpError)
+        .catch((e) => {
+          inspectRequest.error({ json: e });
+          handleHttpError(e);
+        })
         .finally(() => setLoading(false));
     },
-    [coreStart.http, handleHttpError, toastNotifications]
+    [coreStart.http, getRequestInspector, handleHttpError, requestAdapter, toastNotifications]
   );
 
   // Helper function for the graphClientWorkspace to perform a query
   const callSearchNodeProxy = useCallback(
     (indexName: string, query: SearchRequest, responseHandler: GraphSearchCallback) => {
-      const request = {
-        body: JSON.stringify({
-          index: indexName,
-          body: query,
-        }),
-      };
+      const dsl = { index: indexName, body: query };
+      const request = { body: JSON.stringify(dsl) };
       setLoading(true);
+
+      requestAdapter.reset();
```

Comment:
While here, can we show all the "history" requests that are sent to the server to build the current graph?
Removing this line and the other `reset` few lines below should make it.

I think that might be useful. WDYT @kertal ?

Reply:
It would be ok also a follow up, maybe with some details as changing the `Data` label to something more insightful as (`Search` vs `Expand`).

Reply:
sounds good as a potential follow up 👍